runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/commits/StreamingEngineWorkCommitter.java

```diff
@@ -85,7 +88,7 @@ public static StreamingEngineWorkCommitter create(
   @Override
   @SuppressWarnings("FutureReturnValueIgnored")
   public void start() {
-    if (!commitSenders.isShutdown()) {
+    if (isRunning.compareAndSet(false, true) && !commitSenders.isShutdown()) {
```

Comment:
we do not
was just opting for this route to not use locks

how about just
if isRunning return; 
setting isRunning -> true and have the method be synchronized

I don't think we should throw an exception since we will have to handle it in a caller or else crash

Reply:
ack updating